src/react-native/03-creating-a-new-app/creating-a-new-app.md

```diff
@@ -6,43 +6,161 @@
+
+✏️ Start the virtual device you created in the previous module.
+✏️ Run the command:
+
+```shell
+npm run start
+```
+
+✏️ Push `a` to run the app on the open Android virtual device.
+
+The first time you run it for a given target will take significantly longer than subsequent runs. Once open, you should see the default "Welcome to React Native" screen.
+
+<figure>
+  <img alt="A mobile phone displaying the root page of the newly created application served by the dev server. The page displays a 'Welcome to React Native' header and some miscelaneous instructions that we will be disregarding." src="/static/img/react-native/03-creating-a-new-app/01-generated-app.png" width="600px" />
+  <figcaption style="font-style: italic">
+    Screen capture of the completed exercise.
+  </figcaption>
+</figure>
+
+## Objective 2: Add testing infrastructure
+
+### Testing code with React Testing Library
+
+We will create unit tests for our application using the [React Native Testing Library](https://callstack.github.io/react-native-testing-library/). The unit tests for components will be focused on verifying that a component creates the correct output based on the state the component is provided. We also need to create tests for any React custom Hooks we create to ensure they provide the expected results.
+
+### Setup 2
+
+Before we can create or execute any tests, run the following command to install the packages related to the React Native Testing Library:
+
+✏️ Run:
+
+```shell
+npm install --save-dev @testing-library/react-native@12 @types/jest@29
+```
+
+✏️ Update **jest.config.js** to be:
+
+@diff ../../../exercises/react-native/03-creating-a-new-app/01-solution/jest.config.js ../../../exercises/react-native/03-creating-a-new-app/02-solution/jest.config.js
+
+✏️ Create **jest-setup.ts** and update it to be:
+
+@sourceref ../../../exercises/react-native/03-creating-a-new-app/02-solution/jest-setup.ts
+
+✏️ Delete the unneeded generated files inside `__tests__`.
+
+✏️ Create **App.test.tsx** and update it to be:
+
+@sourceref ../../../exercises/react-native/03-creating-a-new-app/02-solution/App.test.tsx
+
+✏️ Replace the contents of **App.tsx** and update it to be:
+
+@sourceref ../../../exercises/react-native/03-creating-a-new-app/02-solution/App.tsx
+
+### Verify 2
+
+After completing the previous steps you will be able to run unit tests. Execute the command:
+
+```shell
+npm run test
+```
+
+and "Test Suites: 1 passed, 1 total" will be written to the console.
+
+## Objective 3: Cleanup the generated code and add helpers
+
+Before we begin adding any content, its a good idea to clean up generated files and add code quality tools.
```

Comment:
`its` -> `it’s`

Reply:
I thought i had the grammarly plugin installed...